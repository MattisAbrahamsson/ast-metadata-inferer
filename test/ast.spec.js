import fs from 'fs';
import path from 'path';
import AstNodeTypeVerifier from '../src';

jest.setTimeout(2000000);

describe('AstNodeTypeVerifier', () => {
  it('should return basic output', async () => {
    const [record, ...records] = await AstNodeTypeVerifier();
    expect(record).toHaveProperty('apiType');
    expect(record).toHaveProperty('type');
    expect(record).toHaveProperty('protoChain');
    expect(record).toHaveProperty('protoChainId');
    expect(record).toHaveProperty('astNodeType');
    expect(record).toHaveProperty('isStatic');
    // expect(record).toMatchSnapshot();

    records.forEach((_record) => {
      expect(_record).toHaveProperty('apiType');
      expect(_record).toHaveProperty('type');
      expect(_record).toHaveProperty('protoChain');
      expect(_record).toHaveProperty('protoChainId');
      expect(_record).toHaveProperty('astNodeType');
      expect(_record).toHaveProperty('isStatic');
    });

    const filepath = path.join(__dirname, '..', 'meta.json');
    const file = await fs.promises.readFile(filepath);

    const recordsCount = JSON.parse(file.toString()).length;
    expect(recordsCount).toBeGreaterThanOrEqual(8000);
    expect(recordsCount.length).toMatchSnapshot();

    console.log(`${recordsCount} records in meta.json`);
  }, 1000000000);
});
