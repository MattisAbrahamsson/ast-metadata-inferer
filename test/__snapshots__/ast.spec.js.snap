// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AstMetadataInferer should return basic output 1`] = `
Object {
  "apiType": "js-api",
  "astNodeType": Array [
    "NewExpression",
  ],
  "isStatic": true,
  "protoChain": Array [
    "AnalyserNode",
  ],
  "protoChainId": "AnalyserNode",
  "type": "js-api",
}
`;

exports[`AstMetadataInferer should return basic output 2`] = `
Object {
  "apiType": "js-api",
  "astNodeType": Array [
    "NewExpression",
  ],
  "isStatic": true,
  "protoChain": Array [
    "AnalyserNode",
  ],
  "protoChainId": "AnalyserNode",
  "type": "js-api",
}
`;

exports[`AstMetadataInferer should return basic output 3`] = `6015`;

exports[`AstMetadataInferer should write to meta.json correctly 1`] = `
Object {
  "apiType": "js-api",
  "astNodeType": Array [
    "NewExpression",
  ],
  "isStatic": true,
  "protoChain": Array [
    "AnalyserNode",
  ],
  "protoChainId": "AnalyserNode",
  "type": "js-api",
}
`;

exports[`AstMetadataInferer should write to meta.json correctly 2`] = `5963`;
